core/src/main/java/org/elasticsearch/index/query/functionscore/DecayFunctionParser.java

```diff
@@ -66,14 +64,7 @@
  * parameters origin and scale.
  * <p>
  * To write a new scoring function, create a new class that inherits from this
- * one and implements {@link #getBuilderPrototype()} and {@link #getNames()}.
- * Also create its corresponding {@link DecayFunctionBuilder}. The latter needs to
- * implement {@link DecayFunctionBuilder#doReadFrom(StreamInput)} and
- * {@link DecayFunctionBuilder#doWriteTo(StreamOutput)} for serialization purposes,
- * {@link DecayFunctionBuilder#doEquals(DecayFunctionBuilder)} and
- * {@link DecayFunctionBuilder#doHashCode()} for equality checks,
- * {@link DecayFunctionBuilder#getName()} that returns the name of the function and
- * {@link DecayFunctionBuilder#getDecayFunction()} which returns the corresponding lucene function.
```

Comment:
They became inaccurate. So I said "implement the thing" and then added
javadocs to the thing to implement. I think that is less likely to become
out of date?
On Mar 25, 2016 5:20 PM, "Luca Cavanna" <notifications@github.com> wrote:

> In
> core/src/main/java/org/elasticsearch/index/query/functionscore/DecayFunctionParser.java
> <https://github.com/elastic/elasticsearch/pull/17351#discussion_r57492451>
> :
>
> > @@ -66,14 +64,7 @@
> >   * parameters origin and scale.
> >   * <p>
> >   * To write a new scoring function, create a new class that inherits from this
> > - * one and implements {@link #getBuilderPrototype()} and {@link #getNames()}.
> > - * Also create its corresponding {@link DecayFunctionBuilder}. The latter needs to
> > - * implement {@link DecayFunctionBuilder#doReadFrom(StreamInput)} and
> > - * {@link DecayFunctionBuilder#doWriteTo(StreamOutput)} for serialization purposes,
> > - * {@link DecayFunctionBuilder#doEquals(DecayFunctionBuilder)} and
> > - * {@link DecayFunctionBuilder#doHashCode()} for equality checks,
> > - * {@link DecayFunctionBuilder#getName()} that returns the name of the function and
> > - * {@link DecayFunctionBuilder#getDecayFunction()} which returns the corresponding lucene function.
>
> was all this javadocs too much?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/17351/files/257ec4587d6121483ac270ce780c03556c88c53e#r57492451>
>